workbench-core/accounts/src/services/projectService.ts

```diff
@@ -80,4 +71,220 @@ export default class ProjectService {
+
+    // Create ProjectAdmin and Researcher groups--TODO implement after dynamic AuthZ
+    // const createProjectAdminGroupResponse = this._dynamicAuthorizationService.createGroup(user, {
+    //   groupId: `${projId}#PA`,
+    //   description: `Project Admin group for ${projId}`
+    // });
+    // const createResearcherGroupResponse = this._dynamicAuthorizationService.createGroup(user, {
+    //   groupId: `${projId}#Researcher`,
+    //   description: `Researcher group for ${projId}`
+    // });
+    // if (!createProjectAdminGroupResponse.created || !createResearcherGroupResponse.created) {
+    //   throw Boom.badImplementation(
+    //     'Failed to create Project Admin group or Researcher group with dynamic authorization service.'
+    //   );
+    // }
+
+    // Create Permissions for the groups--TODO implement after dynamic AuthZ
+    // const identityPermissions: IdentityPermission[] = this._createIdentityPermissionsForProject(projectId);
+    // const createIdentityPermissionsResponse = this._dynamicAuthorizationService.createIdentityPermissions(
+    //   user,
+    //   identityPermissions
+    // );
+    // if (!createIdentityPermissionsResponse.created) {
+    //   throw Boom.badImplementation(
+    //     'Failed to create batch identity permissions for project with dyamic authorization service.'
+    //   );
+    // }
+
+    // Create Proj in DDB
+    const currentTime = new Date().toISOString();
+
+    const newProject: Project = {
+      id: projectId,
+      createdAt: currentTime,
+      costCenterId: params.costCenterId,
+      description: params.description,
+      name: params.name,
+      updatedAt: currentTime,
+      status: 'AVAILABLE',
+      // Acc Metadata (get from cost center)
+      subnetId: costCenter.subnetId,
+      vpcId: costCenter.vpcId,
+      envMgmtRoleArn: costCenter.envMgmtRoleArn,
+      externalId: costCenter.externalId,
+      encryptionKeyArn: costCenter.encryptionKeyArn,
+      environmentInstanceFiles: costCenter.environmentInstanceFiles,
+      hostingAccountHandlerRoleArn: costCenter.hostingAccountHandlerRoleArn,
+      awsAccountId: costCenter.awsAccountId,
+      accountId: costCenter.accountId
+    };
+    try {
+      await this._aws.helpers.ddb
+        .update(buildDynamoDBPkSk(projectId, resourceTypeToKey.project), {
+          item: this._formatForDDB(newProject)
+        })
+        .execute();
+    } catch (e) {
+      console.error('Failed to create project', e);
+      throw Boom.internal('Failed to create project');
+    }
+
+    return newProject;
+  }
+
+  private _formatForDDB(project: Project): { [key: string]: string } {
+    const dynamoItem: { [key: string]: string } = {
+      ...project,
+      resourceType: 'project',
+      depdency: project.accountId
+    };
+
+    delete dynamoItem.accountId;
+
+    return dynamoItem;
+  }
+
+  private async _isProjectNameInUse(projectName: string): Promise<void> {
+    // query by name
```

Comment:
what about naming this `const nameQueryResponse` so we don't need the comment?

Reply:
This was supposed to be helpful :)